core/src/main/java/org/elasticsearch/indices/breaker/StaticCircuitBreakerService.java

```diff
@@ -26,14 +26,30 @@
 /**
  * Class that returns a breaker that never breaks
  */
-public class NoneCircuitBreakerService extends CircuitBreakerService {
+public class StaticCircuitBreakerService extends CircuitBreakerService {
```

Comment:
I found now an even simpler solution by using the real thing (i.e. `HierarchyCircuitBreakerService`) and set the breaker limit of the in-flights circuit breaker via settings. I'll revert the implementation of `StaticCircuitBreakerService` and also rename it again to `NoneCircuitBreakerService`.

> PS. just double checking you know you can register an instance with Guice, not only classes.

Yes, I am aware of that.